lib/her/model/base.rb

```diff
@@ -28,6 +28,17 @@ def [](attribute_name)
       def singularized_resource_name
         self.class.name.split('::').last.tableize.singularize
       end
+
+      # @private
+      def initialize_dup(other)
+        @_her_attributes = @_her_attributes.dup
+        @_her_attributes[self.class.primary_key] = nil
+
```

Comment:
To stick w/ [ActiveRecord](https://api.rubyonrails.org/classes/ActiveRecord/Core.html#method-i-dup) behavior, we'll also need to clear or assign `nil` to any association caches. 

Associations which aren't included in the initial request are stored in separate [instance variables](https://github.com/remiprev/her/blob/master/lib/her/model/associations/has_many_association.rb#L20) following a `@_her_association_<name>` convention. 

Associations which are part of the initial request are [merged](https://github.com/remiprev/her/blob/master/lib/her/model/attributes.rb#L88) w/ `@_her_attributes`.